perception/autoware_tensorrt_bevdet/src/preprocess_plugin.cu

```diff
@@ -0,0 +1,347 @@
+{
+  int idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (idx >= dst_h * dst_w * n) return;
+
+  int i = (idx / n) / dst_w;
+  int j = (idx / n) % dst_w;
+  int k = idx % n;
+
+  int pX = (int)roundf((i / radio_h) + offset_h);
+  int pY = (int)roundf((j / radio_w) + offset_w);
+
+  if (pX < src_h && pX >= 0 && pY < src_w && pY >= 0) {
+    int s1 = k * src_img_step + 0 * src_img_step / 3 + pX * src_row_step + pY;
+    int s2 = k * src_img_step + 1 * src_img_step / 3 + pX * src_row_step + pY;
+    int s3 = k * src_img_step + 2 * src_img_step / 3 + pX * src_row_step + pY;
+
+    int d1 = k * dst_img_step + 0 * dst_img_step / 3 + i * dst_row_step + j;
+    int d2 = k * dst_img_step + 1 * dst_img_step / 3 + i * dst_row_step + j;
+    int d3 = k * dst_img_step + 2 * dst_img_step / 3 + i * dst_row_step + j;
+
+    dst_dev[d1] = (static_cast<T>(src_dev[s1]) - static_cast<T>(mean[0])) / static_cast<T>(std[0]);
+    dst_dev[d2] = (static_cast<T>(src_dev[s2]) - static_cast<T>(mean[1])) / static_cast<T>(std[1]);
+    dst_dev[d3] = (static_cast<T>(src_dev[s3]) - static_cast<T>(mean[2])) / static_cast<T>(std[2]);
+  }
+}
+
+namespace nvinfer1
+{
+// class PreprocessPlugin
+PreprocessPlugin::PreprocessPlugin(
+  const std::string & name, int crop_h, int crop_w, float resize_radio)
+: name_(name)
+{
+  m_.crop_h = crop_h;
+  m_.crop_w = crop_w;
+  m_.resize_radio = resize_radio;
+}
+
+PreprocessPlugin::PreprocessPlugin(const std::string & name, const void * buffer, size_t length)
+: name_(name)
+{
+  memcpy(&m_, buffer, sizeof(m_));
+}
+
+PreprocessPlugin::~PreprocessPlugin()
+{
+}
+
+IPluginV2DynamicExt * PreprocessPlugin::clone() const noexcept
+{
+  auto p = new PreprocessPlugin(name_, &m_, sizeof(m_));
+  p->setPluginNamespace(namespace_.c_str());
+  return p;
+}
+
+int32_t PreprocessPlugin::getNbOutputs() const noexcept
+{
+  return 1;
+}
+
+DataType PreprocessPlugin::getOutputDataType(
+  int32_t index, DataType const * inputTypes, int32_t nbInputs) const noexcept
+{
+  // return DataType::kHALF;
+  return DataType::kFLOAT;
+}
+
+DimsExprs PreprocessPlugin::getOutputDimensions(
+  int32_t outputIndex, const DimsExprs * inputs, int32_t nbInputs,
+  IExprBuilder & exprBuilder) noexcept
+{
+  int input_h = inputs[0].d[2]->getConstantValue();
+  int input_w = inputs[0].d[3]->getConstantValue() * 4;
+
+  int output_h = input_h * m_.resize_radio - m_.crop_h;
+  int output_w = input_w * m_.resize_radio - m_.crop_w;
+
+  DimsExprs ret;
+  ret.nbDims = inputs[0].nbDims;
+  ret.d[0] = inputs[0].d[0];
+  ret.d[1] = inputs[0].d[1];
+  ret.d[2] = exprBuilder.constant(output_h);
+  ret.d[3] = exprBuilder.constant(output_w);
+
+  return ret;
+}
+
+bool PreprocessPlugin::supportsFormatCombination(
+  int32_t pos, const PluginTensorDesc * inOut, int32_t nbInputs, int32_t nbOutputs) noexcept
+{
+  bool res;
+  switch (pos) {
+    case 0:  // images
+      res = (inOut[0].type == DataType::kINT8 || inOut[0].type == DataType::kINT32) &&
```

Comment:
The `supportsFormatCombination` function checks for `DataType::kINT8` and `DataType::kINT32` inputs. Ensure this is intentional and covers all expected input data types.  If not, this should be updated. Add a comment explaining the rationale behind these specific types.